internal/parser/grammar.md

```diff
@@ -38,7 +38,7 @@ The parser directly implements the following grammar.
     LogicalAndOperator             = "and" .
     UnaryLogicalExpression         = ComparisonExpression
                                    | UnaryLogicalOperator UnaryLogicalExpression .
-    UnaryLogicalOperator           = "not" .
+    UnaryLogicalOperator           = "not" | "exists" .
```

Comment:
Not sure if exists should be considered a logical operator as it can operate on more than just booleans.

Reply:
Yeah, I see your point. However, it always returns a boolean, so I would say it is used in logical expression 🤔 

I can change that, however, this grammar should exactly match the parser implementation. So, I should also update the parser accordingly.

Reply:
I agree, let's not modify the parser too much with this addition. We'll keep it as you have specified here. I'm resolving this thread.